sql/core/src/main/java/org/apache/spark/sql/sources/v2/writer/WriteBuilder.java

```diff
@@ -70,6 +71,12 @@ default BatchWrite buildForBatch() {
       " does not support batch write");
   }
 
+  /**
+   * Returns a {@link StreamingWrite} to write data to streaming source. By default this method
+   * throws exception, data sources must overwrite this method to provide an implementation, if the
+   * {@link Table} that creates this write returns {@link TableCapability#STREAMING_WRITE} support
+   * in its {@link Table#capabilities()}.
+   */
```

Comment:
Why were these changes not moved to a documentation commit? These are non-functional changes and should not be in this commit.

Reply:
You did the same when you add the batch read/write capability, didn't you? `TableCapability#STREAMING_WRITE` is added in this PR and I have to add related document at the places that need to mention it.

Reply:
I don't see any changes in the [read/write capability PR](https://github.com/apache/spark/pull/24012/files) that changed unrelated files with only documentation, unless the previous version had links to classes that were deleted. You were right that the changes to `Scan` are needed because the links would otherwise break and that PR made a similar change.

But the change here and the change to `Table` are not required and should be done in documentation PRs.